src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/NativeLibraryExtensions.cs

```diff
@@ -0,0 +1,27 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Runtime.InteropServices
+{
+    public static class NativeLibraryExtensions
+    {
+        public static TDelegate GetDelegate<TDelegate>(this NativeLibrary nativeLibrary, string name, bool exactSpelling = false) where TDelegate : class
+        {
+            nativeLibrary.TryGetDelegate<TDelegate>(name, exactSpelling, out var retVal);
+            return retVal;
```

Comment:
Consider having this test-only extension assert that the nullarity of `retVal` and the return value are consistent.